2996_reflection/reflection.md

```diff
@@ -81,21 +110,73 @@ namespace std {
 
 In our initial proposal a value of type `std::meta::info` can represent:
 
-  - an error
+  - an error (corresponding to an "invalid reflection")
   - any (C++) type and type-alias
   - any function or member function
   - any namespace-scope variable or any C++ static data member
   - any non-static data member
   - any constant value
   - any template
+  - any namespace
 
 [ (DV) I'm currently excluding local variable representation.  We can change that if we have a use for them that we feel must be in the initial core functionality. ]
 
+
+The type `std::meta::info` is a _scalar_ type with associated namespace `std::meta`.  Nontype template arguments of type `std::meta::info` are permitted.
+The entity being reflected can affect the linkage of a template instance involving a reflection.  For example:
+
+```c++
+template<auto R> struct S {};
+
+extern int x;
+static int y;
+
+S<^x> sx;  // S<^x> has external name linkage.
+S<^y> sy;  // S<^y> has internal name linkage.
+```
+
+
 ## The Reflection Operator (`^`)
 
+The reflection operator produces a reflection value from a grammatical construct (its operand):
+
+> | _unary-expression_:
+> |       ...
+> |       `^` `::`
+> |       `^` _namespace-name_
+> |       `^` _type-id_
+> |       `^` _cast-expression_
+
+Note that _cast-expression_ includes _id-expression_, which in turn can designate templates, member names, etc.
+
+The current proposal requires that the _cast-expression_ be:
+
+  - a _primary-expression_ referring to a function or member function, or
+  - a _primary-expression_ referring to a namespace-scope variable or to a static data member, or
+  - a _primary-expression_ referring to a nonstatic data member, or
+  - a _primary-expression_ referring to a template, or
+  - a constant-expression.
```

Comment:
I think here let's add some examples of things that you _cannot_ reflect on (in the current proposal). Cause outside of local variable, I can't immediately think of something.